### ISSUE TITLE
cartographer_node does not publish Poses

### ISSUE THREAD

**Issue by rotu** [ros2/cartographer_ros#30]
The `cartographer_node` only publishes the location of the robot relative to the map as a Transform, and does not publish the robot position as `geometry_msgs.msg.PoseWithCovariance`. It would be nice to have it as a pose for purposes of navigation and/or sensor fusion.

**Comment by clalancette** [comment]
> The `cartographer_node` only publishes the location of the robot relative to the map as a Transform, and does not publish the robot position as `geometry_msgs.msg.PoseWithCovariance`. It would be nice to have it as a pose for purposes of navigation and/or sensor fusion.

Seems reasonable.  As far as I can tell, this is a decision that upstream cartographer_ros has made, so I'd suggest opening a PR there; we'll inherit it the next time we do an update.

**Comment by rotu** [comment]
There is no suitable upstream cartographer_ros branch using ROS2. Perhaps there should be?

**Comment by clalancette** [comment]
> There is no suitable upstream cartographer_ros branch using ROS2. Perhaps there should be?

Yes, there should be.  It essentially entails someone working with upstream cartographer_ros to propose a design document, respond to comments, and do the work to start feeding the port to whatever is decided there.  Unfortunately, I do not have the time to do this work, but I would be happy to provide some side support and guidance for someone who wants to take this on.

In the meantime, another viable path would be to add the PoseWithCovariance publication you want to the upstream master (ROS 1) branch.  It would then be a straightforward piece of the 'port' that we are maintaining here.
